### ISSUE TITLE
[Feat] Change "defaultProps" strategy in components to "JavaScript default parameters"

### ISSUE THREAD

**Issue by luanmm** [visgl/react-map-gl#2117]
### Target Use Case

Anticipate to deprecated functionality removal (and avoid error/warning messages in console).

### Proposal

As it is stated in React DOM warnings:

```
Warning: ScaleControl: Support for defaultProps will be removed from function components in a future major release. Use JavaScript default parameters instead.
```

Components should be using JavaScript default parameters as an alternative.

**Comment by elewin** [comment]
I am getting this on Marker as well:

```
Warning: Marker: Support for defaultProps will be removed from function components in a future major release. Use JavaScript default parameters instead
``

This is using react-map-gl 7.0.25

**Comment by luanmm** [comment]
Looking in v7.0.25 tag, seems that PR #2173 is not present in code. Don't know why exactly, as it appears to be merged. So it should keep giving warnings, indeed, until a release with this change.

**Comment by Pessimistress** [comment]
This will be landed in v7.1z
